### DISCUSSION TITLE
java8/optional

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#80] [Announcements]
# java8/optional

Java程序员进阶之路，小白的零基础Java教程，从入门到进阶，Java 8 Optional最佳指南

https://tobebetterjavaer.com/java8/optional.html

**Comment by wanglililian** [comment]
写的好棒！！要是最后还能有一个更复杂更具实用性的的例子就好啦~~

**Comment by wylgithub** [comment]
确实现年过三十的程序员基本上都是在吃老本。很少有人有兴趣，有精力停下来认真学习和实践了！可以平凡但是拒绝平庸！共勉！！！

**Comment by f3tknco** [comment]
您好，请问 [04、判断值是否存在](https://tobebetterjavaer.com/java8/optional.html#_04、判断值是否存在) 中，关于 `Java 11 后还可以…` 这里的例子，是否应该为

System.out.println(opt.`isEmpty`());

**Comment by KayunZhang** [comment]
二哥  最后那段代码，如果要求密码不能为password的话  可能Predicate<String> eq = pwd -> pwd.equals("password");这段代码要改成Predicate<String> eq = pwd -> !pwd.equals("password");例子中返回false才对吧
